pitest-entry/src/test/java/org/pitest/mutationtest/incremental/IncrementalAnalyserTest.java

```diff
@@ -34,23 +54,41 @@
   @Mock
   private CoverageDatabase    coverage;
 
+  private LogCatcher logCatcher;
+
   @Before
   public void setUp() {
+    configureLogCatcher();
+
     MockitoAnnotations.initMocks(this);
     this.testee = new IncrementalAnalyser(this.history, this.coverage);
   }
 
+  private void configureLogCatcher() {
+    logCatcher = new LogCatcher();
+    final Logger logger = Log.getLogger();
+
+    for (Handler handler : logger.getHandlers()) {
+      logger.removeHandler(handler);
+    }
+    logger.addHandler(logCatcher);
```

Comment:
@StefanPenndorf Are we introducing a memory leak into the test suite here? If nothing removes the LogCatcher when the last runs then presumably it continues to accumulate log entries until the JVM exits?

Reply:
Yes, my first commit had this issue: It introduced a memory leak as well as changing the behaviour for subsequent tests. Thus I decided not to disable default handlers by removing them and to remove the log catcher in a tear down method (annotated with ```@After```)

I guess you found the fix because you merged my change in anyway.

Reply:
Saw your change come in so merged it - hopefully not prematurely?

Tracked down a leak exactly like this one in a commercial test suite years ago, so knew to look for it when testing logging.

Reply:
Everything well, pull request was complete with two commits. Would have left a note if I knew there was something wrong/missing.